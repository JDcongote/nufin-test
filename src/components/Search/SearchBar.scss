@import '../../styles/variables/index';
@import'../../styles/tools/mixins';

.search-bar {
    display: flex;
    justify-content: center;

    @include breakpoint(tablet-portrait-up) {
        width: 100%;
    }

    &__input {
        margin: 10px 1em;
        border-radius: var(--border--radius--normal);
        border: none;
        padding: 0.5em 1em;
        outline: none;
        width: 100%;

        @include breakpoint(phone-up) {
            margin: 10px 1em;
        }

        &::placeholder {
            opacity: 0.5;
        }
    }
}