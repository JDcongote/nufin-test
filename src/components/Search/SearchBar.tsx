import React from 'react';
import './SearchBar.scss';

type SearchBarProps = {
  onSearch: (filtered: any[]) => void;
  content: any[];
  property: string;
};

type State = {
  searchTerm: string;
};

class SearchBar extends React.PureComponent<SearchBarProps, State> {
  componentDidMount() {
    this.setState({ searchTerm: '' });
  }

  doSearch(term: string) {
    if (term !== '') {
      const filtered = this.props.content.filter(
        item => item[this.props.property].toLowerCase().indexOf(term) > -1
      );
      this.props.onSearch(filtered);
    } else {
      this.props.onSearch(this.props.content);
    }
  }

  onInputChange(event: React.BaseSyntheticEvent) {
    const term = event.target.value.toLowerCase();
    this.setState({ searchTerm: term });
    this.doSearch(term);
  }
  render() {
    return (
      <div className="search-bar">
        <input
          className="search-bar__input"
          type="text"
          placeholder="Search"
          onChange={this.onInputChange.bind(this)}
        ></input>
      </div>
    );
  }
}

export default SearchBar;
