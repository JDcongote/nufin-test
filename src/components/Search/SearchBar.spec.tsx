import SearchBar from './SearchBar';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchBar', () => {
  it('Basic Search Bar Test', () => {
    const content = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const onSearch = jest.fn();
    const property = 'name';

    const render = renderer.create(
      <SearchBar content={content} onSearch={onSearch} property={property} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('Search test', () => {
    const content = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const onSearch = val => {
      expect(val.length).toBeGreaterThan(0);
      expect(val.name).toBe('a');
    };
    const property = 'name';

    const wrapper = mount(
      <SearchBar content={content} onSearch={onSearch} property={property} />
    );
    wrapper.find('.search-bar__input').simulate('keydown', { which: 'a' });
  });
});
