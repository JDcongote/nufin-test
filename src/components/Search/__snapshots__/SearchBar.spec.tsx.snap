// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Basic Search Bar Test 1`] = `
<div
  className="search-bar"
>
  <input
    className="search-bar__input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
</div>
`;
